common/browser.go

```diff
@@ -429,11 +424,16 @@ func (b *Browser) Contexts() []api.BrowserContext {
 	return contexts
 }
 
+// IsConnected returns whether the WebSocket connection to the browser process
+// is active or not.
 func (b *Browser) IsConnected() bool {
-	b.connectedMu.RLock()
-	defer b.connectedMu.RUnlock()
-
-	return b.connected
+	var ok bool
+	select {
+	case _, ok = <-b.browserProc.lostConnection:
```

Comment:
Instead of checking it like this, we can hand over this responsibility to the browser process:

```go
Browser.IsConnnected() bool {
  return b.browserProc.connected()
}
```

Reply:
Good idea, will do :+1:

Reply:
Resolved by 0f468ea.